src/main/java/org/zalando/nakadi/service/subscription/zk/ZkSubscriptionNode.java

```diff
@@ -39,23 +38,24 @@ public void setSessions(final Session[] sessions) {
         return sessions;
     }
 
-    public Partition.State guessState(final String partition) {
-        return getPartitionWithActiveSession(partition).map(Partition::getState).orElse(Partition.State.UNASSIGNED);
+    public Partition.State guessState(final String eventType, final String partition) {
+        return getPartitionWithActiveSession(eventType, partition)
+                .map(Partition::getState)
+                .orElse(Partition.State.UNASSIGNED);
     }
 
-    private Optional<Partition> getPartitionWithActiveSession(final String partition) {
+    private Optional<Partition> getPartitionWithActiveSession(final String eventType, final String partition) {
         return Stream.of(partitions)
-                .filter(p -> p.getPartition().equals(partition))
+                .filter(p -> p.getPartition().equals(partition) && p.getEventType().equals(eventType))
                 .filter(p -> Stream.of(sessions).anyMatch(s -> s.getId().equalsIgnoreCase(p.getSession())))
                 .findAny();
     }
 
     @Nullable
-    public String guessStream(final String partition) {
-        return getPartitionWithActiveSession(partition).map(Partition::getSession).orElse(null);
+    public String guessStream(final String eventType, final String partition) {
+        return getPartitionWithActiveSession(eventType, partition)
+                .map(Partition::getSession)
+                .orElse(null);
```

Comment:
Could you return empty string here? In that case you won't need Optional at SubscriptionService.java:270

Reply:
For me this is actually an anti-pattern to return empty String when there is no value. That's why I would not like to change it. We replace `null` with empty String before showing that to user - which is more or less fine. But I think this method should not return empty String to indicate no value.

Reply:
Returning `null` is even worse! Return optional then :)

Reply:
No, I think returning `null` is much better than returning empty String to indicate that there is no value.
Also, the purpose of my PR was a bugfix, not a refactoring. I don't really understand why I should refactor here :)